### ISSUE TITLE
bag.to_dataframe() not working with topk()

### ISSUE THREAD

**Issue by jslmann** [dask/dask#2474]
import bag from dask as db
db.from_sequence([(i,i**2) for i in range(5)]).topk(3, lambda x: x[1]).to_dataframe().compute()
>>> ValueError: DataFrame constructor not properly called!

CPython 3.6.1
IPython 6.1.0
dask.__version__ = 0.15.0

**Comment by martindurant** [comment]
It seems to occur here:
```
> /Users/mdurant/Downloads/dask/dask/bag/core.py(1992)to_dataframe()
   1989
   1990 def to_dataframe(seq, columns, dtypes):
   1991     import pandas as pd
-> 1992     res = pd.DataFrame(reify(seq), columns=list(columns))
   1993     return res.astype(dtypes, copy=False)
```

where wrapping `reify` (which produces tuples) in `list` solves the problem. I don't know if that is the right solution.

**Comment by jcrist** [comment]
Thanks for the issue report. This is fixed by #2476.

```python
In [1]: import dask.bag as db

In [2]: db.from_sequence([(i,i**2) for i in range(5)]).topk(3, lambda x: x[1]).to_dataframe().compute()
Out[2]:
   0   1
0  4  16
1  3   9
2  2   4
```
